### PR TITLE
GitHub Bot

### PR DESCRIPTION
**Pull request by Garmelon**
This PR adds a GitHub bot that responds to PR comments containing only the text `!bench` by benchmarking the PR's current state and posting a link to the results. This requires a small change to the config file since the bot needs to know the base URL it should use for links.

To enable GitHub bot functionality for a repo, create a personal access token for the account the bot should use and set it via the repo detail page. The token only requires the `public_repo` scope to leave reactions and comments. No other scopes are required.

@Kha We have a small design question: Should the bot respond to all `!bench` commands or should it only listen to contributors/owners/...?

If the bot has no verification, it would be possible for anybody to fill the queue with tasks simply by creating lots of PRs and commenting `!bench` on each of them, or by creating a single PR and alternating between commenting `!bench` and pushing a new commit to the PR.

However...
1. If somebody is spamming PRs or comments, that's already a problem in and of itself
2. The queue can always be cleaned up manually
3. Commits that have already been benchmarked will not get re-benchmarked by `!bench` commands. Multiple `!bench` commands for the same commit will not cause multiple runs.

In conclusion, it seems like allowing anybody with commenting privileges to also start a benchmark is probably fine?

### CODE REVIEW

**Comment by Kha** [comment]
> Should the bot respond to all !bench commands or should it only listen to contributors/owners/...?

That's a good question. Since the build is not done inside any sandbox, it should probably be restricted to users with merge rights.

**Comment by Kha** [comment]
By the way, the bot will post a link to the comparison page when finished, correct? Would it also be possible to include a quick overview of the significant results (just percentages), or lack thereof, before the link?

**Comment by Garmelon** [comment]
At the moment, it will link to the run's detail page, which includes a comparison to the previous run (which will usually not exist for PR commits). It would probably make more sense to try to compare against the commit where the PR branched off (which may fail with some PR structures), or against the latest commit on the tracked branch (which may fail if multiple branches are tracked).

Once we've chosen a commit to compare against, including a short summary like that is possible.

**Comment by Kha** [comment]
> It would probably make more sense to try to compare against the commit where the PR branched off 

Yes, this looks like the most reasonable choice. If there are no measurements for that commit, users can still select a different comparison target manually.

**Comment by Garmelon** [comment]
I think this PR should cover everything we talked about now, so it's probably time to test on our aaaaaaah instance.
